### PR TITLE
Add Multicast protocol to implement multi-delegate systems.

### PR DESCRIPTION
**Pull request by Tom-Le**
Lifted this from a current project 'cause I think it could be useful in other projects. This lets us quickly create classes that can have multiple delegates simply by conforming to `Multicast`.

### CODE REVIEW

File: SwiftWisdom/Core/Multicast.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
+//
+//  Multicast.swift
+//  SwiftWisdom
+//
+//  Created by Logan Wright on 9/4/15.
```
Comment (alexpersian) [inline SwiftWisdom/Core/Multicast.swift]: Logan made this?

Comment (Tom-Le) [reply SwiftWisdom/Core/Multicast.swift]: Yeah, more or less. I only made slight modifications and added doc comments.

**Review by paulrolfe — APPROVED** [review]
:shipit:

File: SwiftWisdom/Core/CommonTypes/Multicast.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public weak var weak: T?
+
+    public init(_ weak: T) {
+        self.weak = weak
+    }
+}
+
+/**
+    Conforming to this protocol lets a class implement a multi-delegate system.
+
```
Comment (brightredchilli) [inline SwiftWisdom/Core/CommonTypes/Multicast.swift:24]: lots of additional spaces between comments...did you mean to leave them in?

File: SwiftWisdomTests/CommonTypes/MulticastTests.swift [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            }
+        }
+
+        let broadcaster = Broadcaster()
+        broadcaster.add(delegate: subscriberA)
+        broadcaster.add(delegate: subscriberB)
+        broadcaster.add(delegate: subscriberC)
+        broadcaster.add(delegate: subscriberA)
+        broadcaster.broadcast(that: "We are intrepid") // get it?
+        if !messageRecipients.ip_contains(all: ["A", "B", "C"]) || messageRecipients.count != 3 {
```
Comment (brightredchilli) [inline SwiftWisdomTests/CommonTypes/MulticastTests.swift]: I think we should prefer the positive assertion here - that messageRecipients contains those items, and the count is three.

File: SwiftWisdomTests/CommonTypes/MulticastTests.swift [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            if message == "We are intrepid" {
+                messageRecipients.append("C")
+            }
+        }
+
+        let broadcaster = Broadcaster()
+        broadcaster.add(delegate: subscriberA)
+        broadcaster.add(delegate: subscriberB)
+        broadcaster.add(delegate: subscriberC)
+        broadcaster.add(delegate: subscriberA)
```
Comment (brightredchilli) [inline SwiftWisdomTests/CommonTypes/MulticastTests.swift:93]: I would also suggest that this particular test be broken out - this test should be named `testAddOnSameDelegateShouldNotReadd`. 

The overall test should be named `testBroadcastShouldBroadcastToAllDelegates`

File: SwiftWisdomTests/CommonTypes/MulticastTests.swift [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                messageRecipients.append("C")
+            }
+        }
+
+        let broadcaster = Broadcaster()
+        broadcaster.add(delegate: subscriberA)
+        broadcaster.add(delegate: subscriberB)
+        broadcaster.add(delegate: subscriberC)
+        broadcaster.removeDelegates { $0 === subscriberA || $0 === subscriberC }
+        broadcaster.broadcast(that: "We are gorgeous")
```
Comment (brightredchilli) [inline SwiftWisdomTests/CommonTypes/MulticastTests.swift:152]: love these messages.

**Review by brightredchilli — CHANGES_REQUESTED** [review]

File: SwiftWisdom/Core/CommonTypes/Multicast.swift (hunk 2 above)
Comment (Tom-Le) [reply SwiftWisdom/Core/CommonTypes/Multicast.swift:24]: Yeah, I don't mind deleting them but I thought it looked more readable

**Review by brightredchilli — APPROVED** [review]

**Pull request merged**